evm_trace/base.py

```diff
@@ -54,3 +56,121 @@ class CallTreeNode(BaseModel):
     @validator("address", "calldata", "returndata", pre=True)
     def validate_hexbytes(cls, v) -> HexBytes:
         return _convert_hexbytes(cls, v)
+
+
+def get_calltree_from_trace(
+    trace: Iterator[TraceFrame], show_internal=False, **root_node_kwargs
+) -> CallTreeNode:
+    """
+    Creates a CallTreeNode from a given transaction trace.
+
+    Args:
+        trace (Iterator[TraceFrame]): Iterator of transaction trace frames.
+        show_internal (bool): Boolean whether to display internal calls. Defaulted to False.
+        root_node_kwargs (dict): Keyword argments passed to the root ``CallTreeNode``.
+
+    Returns:
+        CallTreeNode: Call tree of transaction trace.
+    """
+
+    return _create_node_from_call(
+        trace=trace,
+        show_internal=show_internal,
+        **root_node_kwargs,
+    )
+
+
+def _extract_memory(offset: HexBytes, size: HexBytes, memory: List[HexBytes]) -> HexBytes:
+    """
+    Extracts memory from the EVM stack.
+
+    Args:
+        offset (HexBytes): Offset byte location in memory.
+        size (HexBytes): Number of bytes to return.
+        memory (List[HexBytes]): Memory stack.
+
+    Returns:
+        HexBytes: Byte value from memory stack.
+    """
+
+    size_int = int(size.hex(), 16)
+
+    if size_int == 0:
+        return HexBytes(b"")
+
+    offset_int = int(offset.hex(), 16)
+
+    # Compute the word that contains the first byte
+    start_word = math.floor(offset_int / 32)
+    # Compute the word that contains the last byte
+    stop_word = math.ceil((offset_int + size_int) / 32)
+
+    byte_slice = b""
+
+    for word in memory[start_word:stop_word]:
+        byte_slice += word
+
+    return HexBytes(byte_slice[(offset_int % 32) : size_int])  # noqa: E203
```

Comment:
If you don't want to use `# noqa: E203`, you can set a variable first:

```python
offset_index = offset_int % 32
return HexBytes(byte_slice[offset_index: size_int])
```

Reply:
Updated with [4f8c476](https://github.com/ApeWorX/evm-trace/pull/4/commits/4f8c4768d925d448403736c8ce76490be1c9e3ca)